src/common/BaseComponent.ts

```diff
@@ -19,9 +19,17 @@ export class BaseComponent<P, S> extends React.Component<P, S> {
   private __events: EventGroup;
   private __disposables: IDisposable[];
 
-  constructor(props?: P) {
+  constructor(props?: P, deprecatedProps?: { [propName: string]: string }) {
```

Comment:
Would be a good idea to add JSDoc to the constructor here, not so much to say too much about the constructor but primarily for the purposes of documenting the `@params`.